app/app/src/main/jni/vsp-pty.cpp

```diff
@@ -139,30 +138,26 @@ extern "C"
 
         jbyteArray serialDataArr = gEnv->NewByteArray(dataSize);
         gEnv->SetByteArrayRegion(serialDataArr, 0, dataSize, (jbyte *)val);
-        jobject object = gEnv->NewObject(serialDataClass, serialDataConstructor, serialDataArr, baudrate);
+        jobject object = gEnv->NewObject(serialDataClass, serialDataConstructor, serialDataArr, static_cast<jint>(baudrate));
 
         gEnv->CallVoidMethod(storeWeakListener, stringCallback, object);
         gEnv->DeleteLocalRef(object);
 
         if (gEnv->ExceptionCheck())
-        {
             gEnv->ExceptionDescribe();
-        }
 
-        if (getEnvStat != JNI_EDETACHED) {
```

Comment:
This is not necessary, DetachCurrentThread guarantees no-op if there's nothing to detach (as attach does)